imagex/imagex_wysiwyg_profiles
# Feature: Improve default profile configuration.

## Ready State: Yes

This pull request does 3 things:
- updates imagex_wysiwyg_profiles feature module to disable the ckeditor profile advanced content filter, which can break the ability to embed media, amongst other things
- updates drupal-org.make to use an updated ckeditor library version and a patched ckeditor module
- adds customization to the ckeditor library via plugin to simplify and improve the editor dialog interfaces

The intention behind this pull request is to move the configuration forward so that we have a more functional default. The most important part of this is ensuring that media can be embedded via the media browser and that that media is embedded using media tokens so the file entities can be tracked. As a working default, there are a lot of known issues that will need to be addressed once this is in place.

This pull request is related to the following JIRA tickets: 
- [ICS-154](https://imagex.atlassian.net/browse/ICS-154)
- [ICS-8](https://imagex.atlassian.net/browse/ICS-8)
- [ICS-7](https://imagex.atlassian.net/browse/ICS-7)
## Requirements & Dependencies
- As a part of updating the imagex_wysiwyg_profiles feature module, some htmlpurifier configuration was removed which appears to be a part of the codehighlighter module. I suspect that this may be a complication for any projects using the codehighlighter module. However, this configuration shouldn't be inimagex_wysiwyg_profiles because it causes the feature to install and remain in a persistent overridden state so long as codehighlighter is not installed. 
- Not a strict requirement, but to properly test, needs https://github.com/imagex/imagex_media/pull/3
## Deployment Instructions
- Merge code.
- Navigate to `imagex_installkit` profile: `cd profiles/imagex_installkit`
- Rebuild ckeditor module: `drush make modules/imagex/modules/imagex_wysiwyg_profiles/drupal-org.make ./ --working-copy --no-gitinfofile --no-core --no-cache --contrib-destination=./ --projects=ckeditor -y`
- Rebuild ckeditor library: `drush make modules/imagex/modules/imagex_wysiwyg_profiles/drupal-org.make ./ --working-copy --no-gitinfofile --no-core  --no-cache --contrib-destination=./ --libraries=ckeditor -y`
- Revert `imagex_wysiwyg_profiles`: `drush fr imagex_wysiwyg_profiles`
- Clear caches with `drush cache-clear all`
## Usage/Test Instructions
- Browse to ckeditor profile configuration /admin/config/content/ckeditor and ensure the Advanced profile is assigned to a text format
- Browse to a content edit form, select the appropriate text format, and embed an image in the editor via the media browser button making sure to enter values in the title and alt attribute fields
- Toggle the editor to plain text mode and back to unbind and rebind the editor
- Verify that the image is being converted to a media token and retains the token
- Save the node and verify the image renders correctly
- Check to ensure that the title and alt attributes were applied correctly to the image in html
- Navigate back to the edit form of the node and toggle the editor to ensure that the media tokens are still applied and in use
- Navigate to the file content section of the site /admin/content/file/list and ensure that the file added to the node is registering the usage in the files "Usage" tab
